### PR TITLE
[CLI] Add "ti example -s/-p/-P" to save/print/pretty_print example source code

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #... (if any)

- [[Click here for the format server]](http://kun.csail.mit.edu:31415/)
- [[Click here for coverage report]](http://codecov.io/gh/taichi-dev/taichi/)

图片

### CODE REVIEW

File: python/taichi/main.py [hunk 1]
```diff
@@ -151,17 +151,53 @@ def example(self, arguments: list = sys.argv[2:]):
...
+            try:
+                # If user have installed `rich`, we can provide syntax highlight:
+                # https://rich.readthedocs.io/en/latest/syntax.html
+                import rich.syntax
+                import rich.console
+                syntax = rich.syntax.Syntax.from_path(target,
+                                                      line_numbers=True)
+                console = rich.console.Console()
+                console.print(syntax)
+            except:
```
Comment (rexwangcc) [inline python/taichi/main.py:188]: would `ModuleNotFoundError, ImportError` be sufficient here?

File: python/taichi/main.py [hunk 2]
```diff
@@ -151,17 +151,62 @@ def example(self, arguments: list = sys.argv[2:]):
...
         sys.path.append(str(examples_dir.resolve()))
-        print(f"Running example {args.name} ...")
 
         # Short circuit for testing
         if self.test_mode: return args
 
+        if args.save:
+            print(f"Saving example {args.name} to current directory...")
+            shutil.copy(target, '.')
+            return
```
Comment (rexwangcc) [inline python/taichi/main.py:188]:
```suggestion
            return 0
```

File: python/taichi/main.py [hunk 3]
```diff
@@ -151,17 +151,62 @@ def example(self, arguments: list = sys.argv[2:]):
...
+
+        if args.pretty_print:
+            try:
+                import rich.syntax
+                import rich.console
+            except ImportError as e:
+                print('To make -P work, please: python3 -m pip install rich')
+                raise e
```
Comment (rexwangcc) [inline python/taichi/main.py:196]: Instead of re-raising the exception, how about return 1?

File: python/taichi/main.py [hunk 4]
```diff
@@ -151,17 +151,62 @@ def example(self, arguments: list = sys.argv[2:]):
...
+            # https://rich.readthedocs.io/en/latest/syntax.html
+            syntax = rich.syntax.Syntax.from_path(target, line_numbers=True)
+            console = rich.console.Console()
+            console.print(syntax)
+            return
```
Comment (rexwangcc) [inline python/taichi/main.py:201]:
```suggestion
            return 0
```

File: python/taichi/main.py [hunk 5]
```diff
@@ -151,17 +151,62 @@ def example(self, arguments: list = sys.argv[2:]):
...
+
+        if args.print:
+            with open(target) as f:
+                print(f.read())
+            return
```
Comment (rexwangcc) [inline python/taichi/main.py:206]:
```suggestion
            return 0
```

File: python/taichi/main.py (hunk 3 above)
Comment (archibate) [reply python/taichi/main.py:196]: Yes, good idea!
```suggestion
                return 1
```

**Pull request merged**